packages/style-engine/src/index.ts

```diff
@@ -27,20 +27,23 @@ export function generate( style: Style, options: StyleOptions ): string {
 	const groupedRules = groupBy( rules, 'selector' );
 	const selectorRules = Object.keys( groupedRules ).reduce(
 		( acc: string[], subSelector: string ) => {
+			const hasSelector = subSelector !== 'undefined';
```

Comment:
Is this line (and the one two lines below) missing a `typeof`?

Reply:
Thanks for raising this. I wasn't sure how to cleanly deal with it.

When there's an undefined object key, `'undefined'` (a string) becomes the key.

```js
const test = { undefined: 2 }
console.log( test['undefined'] ); // 2
```

So... that's why it is how it is 😄

Reply:
Thanks for the explanation! I must have been reading too fast, and glossed over the `Object.keys()` call, and that the `groupedRules` object is a result of calling `groupBy` — as you mention, since `rules` might have a `selector` attribute of `undefined`, it now makes sense why they're grouped under an `undefined` key 👍

This might be a bit nit-picky given that the logic here works well in testing, but I'm wondering if it'd be worth having a separate condition for handling when we want to return inline styles versus selector-based CSS, mostly because we won't want to mix-and-match (e.g. output containing both an inline style _and_ one wrapped in a selector). The idea would be to try to keep the logic for handling inline styles versus selector-based styles a little more separate. I'm imagining something like the following right at the top of the `generate` function:

```TS
	const rules = getCSSRules( style, options );

	// If no selector is provided, treat generated rules as inline styles to be returned as a single string.
	if ( ! options?.selector ) {
		const inlineRules: string[] = [];
		rules.forEach( ( rule ) => {
			inlineRules.push( `${ kebabCase( rule.key ) }: ${ rule.value };` );
		} );
		return inlineRules.join( ' ' );
	}

	const groupedRules = groupBy( rules, 'selector' );

	// ... the rest of the function
```

What do you think?

Reply:
> The idea would be to try to keep the logic for handling inline styles versus selector-based styles a little more separate. 

This makes perfect sense to me. It should makes things easier to read and work with in the future. Thanks for thinking about and for the example!